src/components/ToolMenu/LayerTree/LayerTreeContextMenu/index.tsx

```diff
@@ -123,19 +123,21 @@ export const LayerTreeContextMenu: React.FC<LayerTreeContextMenuProps> = ({
       return;
     }
 
-    const targetFolderName = t('AddLayerModal.externalWmsFolder');
-    const targetGroup: LayerGroup = MapUtil.getLayerByName(map, targetFolderName) as LayerGroup;
-    if (targetGroup) {
-      const existingLayers = targetGroup.getLayers();
-      try {
-        if (existingLayers.getLength() === 1) {
-          targetGroup.set('hideInLayerTree', true);
+    const layerGroups = map.getLayers().getArray();
```

Comment:
This will match all map layers (not only imported and/or external). Is this intended?

Reply:
Yes, because we do not know yet if another layergroup will be added in the future. That's why there is the check in line 162, that looks for the attribute 'isImportedLayer'.

Reply:
agreed, I think there is no other way besides going through each layer in the map.

note that this will just give you layers on the first level, and we only support "custom" layerGroups on this level.